### PR TITLE
Fixed verify in unit tests

### PR DESCRIPTION
**Pull request by edwin-pixel-1**
- Added verify method to view mock object
- Removed unused constant in constants class

### CODE REVIEW

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 1]
```diff
@@ -105,7 +105,7 @@ public void getUserInfoFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onFinishedUser(user);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Use camelCase to rename `OnFinishedListenerCallBack`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public void getUserInfoFromInteractorAndLoadIntoView() {
...
 
-        view.showSubscriberDetails(user);
+        verify(view).showSubscriberDetails(user);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:110]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 3]
```diff
@@ -127,7 +127,7 @@ public void getNullUserInfoRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:130]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 4]
```diff
@@ -149,7 +149,7 @@ public void getNotConnectionUserInfoRequestFromInteractor() {
...
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:152]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 5]
```diff
@@ -173,7 +173,7 @@ public void getRepositoriesListFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
 
-        view.showSubscriberUserRepositories(repoList);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:176]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 6]
```diff
@@ -196,7 +196,7 @@ public void getEmptyRepositoriesListFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showUserError(messageId);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:199]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 7]
```diff
@@ -218,7 +218,7 @@ public void getCancelledRepositoriesListRequestFromInteractor() {
...
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java:221]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 8]
```diff
@@ -89,7 +89,7 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
 
         OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java]: Remane it using camelCase.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 9]
```diff
@@ -89,7 +89,7 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
...
 
-        view.showSubscribersList(subscriberList);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:92]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 10]
```diff
@@ -111,7 +111,7 @@ public void getNotConnectionSubscribersListRequestFromInteractor() {
         OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showSubscribersError(messageId);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:114]: You should verify that the progress indicator is shown.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 11]
```diff
@@ -128,13 +128,12 @@ public void getEmptySubscribersListRequestFromInteractor() {
         presenter.getSubscribersList();
 
         // Then Subscribers list is loaded from Interactor and the callback is captured
-
         verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
 
         OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
 
         int messageId = ErrorMessagesHelper.getMessage(typeError);
-        view.showSubscribersError(messageId);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java:137]: You should verify that the progress indicator is shown.

**Review by lucaslabs — CHANGES_REQUESTED** [review]

**Review by lucaslabs — APPROVED** [review]

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java [hunk 12]
```diff
@@ -51,14 +51,18 @@ public SubscriberDetailPresenterImpl(SubscriberDetailView view, User userData, L
 
     public void onFinishedUser(User userItem) {
         userData = userItem;
-        view.showSubscriberDetails(userData);
+        if (view != null) {
+            view.showSubscriberDetails(userData);
+            view.toggleProgressIndicator(false);
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:55]: The natural order of the experience is: hide the loading then show the data, so, put `view.toggleProgressIndicator(false);` first then show the data.

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java [hunk 13]
```diff
@@ -51,14 +51,18 @@ public SubscriberDetailPresenterImpl(SubscriberDetailView view, User userData, L
...
+        }
     }
 
     @Override
     public void onFailureUser(String errorMessage, ErrorMessagesHelper.TypeError type) {
         int messageId = ErrorMessagesHelper.getMessage(type);
         if (messageId > 0 && view != null) {
             view.showUserError(messageId);
+            view.toggleProgressIndicator(false);
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:64]: Same here.

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java [hunk 14]
```diff
@@ -68,6 +72,7 @@ public void onFailureRepository(String errorMessage, ErrorMessagesHelper.TypeErr
...
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:74]: Same.

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java [hunk 15]
```diff
@@ -76,7 +81,10 @@ public void onFailureRepository(String errorMessage, ErrorMessagesHelper.TypeErr
     public void onFinishedRepository(List<Repository> repositoryList) {
         repositoryListData.clear();
         repositoryListData.addAll(repositoryList);
-        view.showSubscriberUserRepositories(repositoryListData);
+        if (view != null) {
+            view.showSubscriberUserRepositories(repositoryListData);
+            view.toggleProgressIndicator(false);
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/presenter/SubscriberDetailPresenterImpl.java:85]: Same.

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java [hunk 16]
```diff
@@ -108,11 +110,20 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
...
             recyclerViewSubscribers.setAdapter(repositoriesAdapter);
             repositoriesAdapter.setClickListener(this);
+            loadProgressIndicator(getView());
         } else {
             Debug.i(savedInstanceState.toString());
         }
     }
 
+    private void loadProgressIndicator(View view) {
+        progressBar = new ProgressDialog(view.getContext());
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:120]: Why not declaring the Progress indicator in XML?

File: app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/presenter/SubscribersListPresenterImpl.java [hunk 17]
```diff
@@ -71,6 +73,7 @@ public void onFailure(String errorMessage, ErrorMessagesHelper.TypeError type) {
         int messageId = ErrorMessagesHelper.getMessage(type);
         if (messageId > 0 && view != null) {
             view.showSubscribersError(messageId);
+            view.toggleProgress(false);
```
Comment (lucaslabs) [inline app/src/main/java/com/github/globant/githubsubscribers/subscriberslist/presenter/SubscribersListPresenterImpl.java:75]: Hide the progress firts then show the error message.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 18]
```diff
@@ -122,12 +124,14 @@ public void getNullUserInfoRequestFromInteractor() {
         presenter.getUser(USER_NAME_PARAM);
 
         //Then User information is loaded from Interactor and the callback is captured
-        verify(interactor).getUserData(eq(USER_NAME_PARAM), OnFinishedListenerCallBack.capture());
+        verify(interactor).getUserData(eq(USER_NAME_PARAM), OnFinishedListenerCallback.capture());
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureUser(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureUser(errorMessage, typeError);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 19]
```diff
@@ -144,12 +148,14 @@ public void getNotConnectionUserInfoRequestFromInteractor() {
...
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 20]
```diff
@@ -169,11 +175,13 @@ public void getRepositoriesListFromInteractorAndLoadIntoView() {
         presenter.getRepositoryList(USER_NAME_PARAM);
 
         //Then Repositories list is loaded from Interactor and the callback is captured
-        verify(interactor).getUserRepositoryData(eq(USER_NAME_PARAM), OnFinishedListenerCallBack.capture());
+        verify(interactor).getUserRepositoryData(eq(USER_NAME_PARAM), OnFinishedListenerCallback.capture());
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFinishedRepository(repoList);
+        OnFinishedListenerCallback.getValue().onFinishedRepository(repoList);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 21]
```diff
@@ -191,12 +199,14 @@ public void getEmptyRepositoriesListFromInteractor() {
         presenter.getRepositoryList(USER_NAME_PARAM);
 
         //Then Repositories list is loaded from Interactor and the callback is captured
-        verify(interactor).getUserRepositoryData(eq(USER_NAME_PARAM), OnFinishedListenerCallBack.capture());
+        verify(interactor).getUserRepositoryData(eq(USER_NAME_PARAM), OnFinishedListenerCallback.capture());
+        verify(view).toggleProgressIndicator(true);
 
-        OnFinishedListenerCallBack.getValue().onFailureRepository(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailureRepository(errorMessage, typeError);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java [hunk 22]
```diff
@@ -213,12 +223,14 @@ public void getCancelledRepositoriesListRequestFromInteractor() {
...
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscribersdetail/SubscriberDetailPresenterTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 23]
```diff
@@ -85,11 +80,13 @@ public void getSubcriberListFromInteractorAndLoadIntoView() {
         presenter.getSubscribersList();
 
         // Then Subscribers list is loaded from Interactor and the callback is captured
-        verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
+        verify(interactor).getSubscribersDataList(OnFinishedListenerCallback.capture());
+        verify(view).toggleProgress(true);
 
-        OnFinishedListenerCallBack.getValue().onResponse(subscriberList);
+        OnFinishedListenerCallback.getValue().onResponse(subscriberList);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 24]
```diff
@@ -106,12 +103,14 @@ public void getNotConnectionSubscribersListRequestFromInteractor() {
         presenter.getSubscribersList();
 
         // Then Subscribers list is loaded from Interactor and the callback is captured
-        verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
+        verify(interactor).getSubscribersDataList(OnFinishedListenerCallback.capture());
+        verify(view).toggleProgress(true);
 
-        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailure(errorMessage, typeError);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

File: app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java [hunk 25]
```diff
@@ -128,13 +127,14 @@ public void getEmptySubscribersListRequestFromInteractor() {
         presenter.getSubscribersList();
 
         // Then Subscribers list is loaded from Interactor and the callback is captured
+        verify(interactor).getSubscribersDataList(OnFinishedListenerCallback.capture());
+        verify(view).toggleProgress(true);
 
-        verify(interactor).getSubscribersDataList(OnFinishedListenerCallBack.capture());
-
-        OnFinishedListenerCallBack.getValue().onFailure(errorMessage, typeError);
+        OnFinishedListenerCallback.getValue().onFailure(errorMessage, typeError);
```
Comment (lucaslabs) [inline app/src/test/java/com/github/globant/githubsubscribers/subscriberslist/SubscribersListInteractorTest.java]: Remane the variable using camelCase to `onFinishedListenerCallback`.

**Review by lucaslabs — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java (hunk 16 above)
Comment (juliocatano) [reply app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/view/SubscriberDetailFragment.java:120]: +1

File: app/build.gradle [hunk 26]
```diff
@@ -28,7 +28,8 @@ dependencies {
     compile "com.squareup.retrofit2:retrofit:$rootProject.ext.retrofit2Version"
     compile "com.squareup.retrofit2:converter-gson:$rootProject.ext.retrofit2Version"
     compile "com.squareup.picasso:picasso:$rootProject.ext.picassoVersion"
-
+    compile "com.github.d-max:spots-dialog:$rootProject.ext.spotsDialogVersion"
```
Comment (lucaslabs) [inline app/build.gradle]: Please keep it simple and use the progress dialog.

**Review by lucaslabs — CHANGES_REQUESTED** [review]

**Comment by lucaslabs** [comment]
LG

**Review by lucaslabs — APPROVED** [review]

**Pull request merged**